Code/Editor/AssetImporter/AssetImporterManager/AssetImporterManager.cpp

```diff
@@ -129,10 +129,14 @@ bool AssetImporterManager::OnBrowseFiles()
     {
         currentAbsolutePath = QStandardPaths::writableLocation(QStandardPaths::DesktopLocation);
     }
-
     fileDialog.setDirectory(currentAbsolutePath);
 
-    if (!fileDialog.exec())
+    fileDialog.open();
+    while (returnFlag < 0)
+    {
+        QCoreApplication::processEvents(QEventLoop::AllEvents, 100);
+    }
```

Comment:
I don't think this is an ideal fix. This is effectively trying to trick a modal dialog to not be flagged as modal so it won't be auto-dismissed, but still some-what behave as modal. Any other dialog which is exec'd would need similar logic in order to be interacted with in an automated test. We should either:

1. Change this dialog to not be modal, unless it needs to be for the workflow to behave properly

or

2. Not include this in the test

The test itself isn't actually verifying the functionality of these options anyways.

Reply:
@cgalvan In the Qt documentation, the recommendation is **not** to use exec() but open() with modal dialogs as exec sets up an internal eventloop that has unspecified side effects. This code is not trying to trick anything, but I suggest should be the correct way to use modals. This is not my test, just a fix for Jenkins not working the same as local code.